wizeline/wizeline-bot-broker-logger
# BotLog model validation

#### What this PR does?
- Defines a model for the incomming logs
- Add validation layer to avoid enqueueing bad logs into the analytics stream
- Documentation

#### Where should the reviewer start?
Put special atention reviewing the schema. We need to make sure it is alright and it is what we are expecting.

#### How should this be manually tested?
- Set your AWS credentials as environment variables. [Here](http://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html#cli-environment)
- The Spark application and Postgress should be running already. Make sure you have access to Postgress.
- Run the example: `node examples/single-log-example.js`
- It should not throw any error and after some second there should be results in Postgress
- Mess around with the object being logged only to confirm that the schema works and it throws an error if the object does not follow the schema defined.

#### Any background context you want to provide?
This is going to be used by the backend of any bot we are going to analyze.

#### What are the relevant tickets?
- [TB-17](https://wizeline.atlassian.net/browse/TB-17)
- [TB-43](https://wizeline.atlassian.net/browse/TB-43)

- [x] @juanitodread
